### PR TITLE
cdc: skip prewrite without value

### PR DESCRIPTION
**Pull request by hicqu**
Signed-off-by: qupeng <qupeng@pingcap.com>

### What is changed and how it works?

Issue Number: ref https://github.com/tikv/tikv/issues/12717.
`
What's Changed:

```commit-message
cdc: skip prewrite without value
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Integration test

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/cdc/tests/integrations/test_cdc.rs [hunk 1]
```diff
@@ -2316,3 +2316,39 @@ fn test_resolved_ts_with_learners() {
...
+
+#[test]
+fn test_prewrite_without_value() {
+    let cluster = new_server_cluster(0, 2);
+    cluster.pd_client.disable_default_operator();
+    let mut suite = TestSuiteBuilder::new().cluster(cluster).build();
+    let rid = suite.cluster.get_region(&[]).id;
+    let ctx = suite.get_context(rid);
+    let client = suite.get_tikv_client(rid).clone();
+    let large_value = vec![b'x'; 1024];
```
Comment (overvenus) [inline components/cdc/tests/integrations/test_cdc.rs:2328]:
```suggestion
    let large_value = vec![b'x'; 2*txn_types::SHORT_VALUE_MAX_LEN];
```

File: components/cdc/src/delegate.rs [hunk 2]
```diff
@@ -780,16 +760,20 @@ impl Delegate {
...
                 }
 
-                *occupied = row;
+                let occupied = rows.entry(row.key.clone()).or_default();
+                if occupied.1 {
+                    assert!(!has_value);
+                    has_value = true;
+                    mem::swap(&mut occupied.0.value, &mut row.value);
+                }
+                *occupied = (row, has_value);
```
Comment (overvenus) [inline components/cdc/src/delegate.rs:769]: How about `has_value = !row.value.is_empty()`? So we don't have to change `decode_write` and `decode_default`.

File: components/cdc/src/delegate.rs [hunk 3]
```diff
@@ -481,83 +481,63 @@ impl Delegate {
         let mut rows = vec![Vec::with_capacity(entries_len)];
         let mut current_rows_size: usize = 0;
         for entry in entries {
+            let (mut row, mut _has_value) = (EventRow::default(), false);
```
Comment (hi-rustin) [inline components/cdc/src/delegate.rs:484]: Why does `_has_value` have a `_` prefix?

File: components/cdc/src/delegate.rs (hunk 2 above)
Comment (hi-rustin) [reply components/cdc/src/delegate.rs:769]: This makes sense to me.

**Review by hi-rustin — COMMENTED** [review]
rest LGTM

File: components/cdc/src/delegate.rs (hunk 2 above)
Comment (hi-rustin) [reply components/cdc/src/delegate.rs:769]: +1 👍

**Review by overvenus — APPROVED** [review]

File: components/cdc/src/delegate.rs (hunk 2 above)
Comment (hicqu) [reply components/cdc/src/delegate.rs:769]: We have discussed it offline. If there is an empty short value, `has_value` should also be `true` but `row.value` will be empty.

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**